### ISSUE TITLE
`nullspace` incorrect for single column non-zero matrix

### ISSUE THREAD

**Issue by JeffFessler** [JuliaLang/julia#50475]
Julia 1.9.2:

```julia
using LinearAlgebra
nullspace(ones(3,1))
```
the output is `1×0 Matrix{Float64}` suggesting that the null space is empty, but it is not empty,
it consists solely of the zero element.
The mathematically correct output should be:
`zeros(1,1)` i.e. `1×1 Matrix{Float64}: 0.0`

btw, sadly the blue links to "source" code in the docs have disappeared in the julia manual for the LinearAlgebra functions, e.g.,
https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.nullspace

**Comment by stevengj** [comment]
The current behavior is correct, I believe.

The `nullspace` function returns a *basis* for the nullspace.   When the nullspace is 0-dimensional, consisting only of the zero vector, then the correct basis is indeed the empty basis (since the empty sum is zero).  That is the case here.

**Comment by stevengj** [comment]
Re-opening because I think that the correct answer here is arguably a 3x0 matrix, not a 1x0 matrix.

**Comment by JeffFessler** [comment]
I apologize for this embarrassing error on my part.
This came up because a student asked about a related topic and I reminded to the student that a basis cannot have any zero vectors in it.  Then I foolishly made the same mistake here!
I think what threw me is the `1×0 Matrix` output which did not seem synonymous with the empty set.  I was expecting Julia to return `∅`.  I edited the title just so that the issue looks less dumb.  😅 
I am not proposing to reopen the issue.  Returning an empty matrix is fine.  Thanks!
